bmwqemu.pm

```diff
@@ -198,24 +197,24 @@ sub update_line_number {
 }
 
 # pretty print like Data::Dumper but without the "VAR1 = " prefix
-sub pp {
+sub pp(@args) {
     # FTR, I actually hate Data::Dumper.
-    my $value_with_trailing_newline = Data::Dumper->new(\@_)->Terse(1)->Useqq(1)->Dump();
+    my $value_with_trailing_newline = Data::Dumper->new(\@args)->Terse(1)->Useqq(1)->Dump();
     chomp($value_with_trailing_newline);
     return $value_with_trailing_newline;
 }
 
-sub log_call {
+sub log_call (@args) {
     my $fname = (caller(1))[3];
     update_line_number();
     my $params;
-    if (@_ == 1) {
-        $params = pp($_[0]);
+    if (@args == 1) {
+        $params = pp($args[0]);
     }
     else {
         # key/value pairs
         my @result;
-        my $should_hide = grep { ($_ // '') eq 'secret' } @_;
+        my $should_hide = grep { ($_ // '') eq 'secret' } @args;
         while (my ($key, $value) = splice(@_, 0, 2)) {
```

Comment:
missed the `@_` here: `splice(@_, 0, 2)`

Reply:
I did that and updated all other references of `@_` which includes the TiedHash methods which I previously did not change.